src/full/Agda/Syntax/Concrete/Definitions.hs

```diff
@@ -164,8 +165,18 @@ type NiceTypeSignature  = NiceDeclaration
 data Clause = Clause Name Catchall LHS RHS WhereClause [Clause]
     deriving (Data, Show)
 
+-- | Exception with File and Line source
+data DeclarationException = DeclarationException
+  { deFileLine   :: (String, Int)
```

Comment:
Please introduce a data type for the AgdaSourceErrorLocation (rather than using a pair). E.g.
```haskell
data AgdaSourceErrorLocation = AgdaSourceErrorLocation
  { slocFile :: String  -- ^ File of the Agda source code.
  , slocLine :: Int     -- ^ Line number in this file
  }
```
See the style guide (don't use tuples!).